src/provider/camunda-platform/properties/ExtensionPropertiesProps.js

```diff
@@ -63,27 +65,49 @@ function removeFactory({ commandStack, element, property }) {
       return;
     }
 
+    let values = without(properties.get('values'), property);
+
     commands.push({
-      cmd: 'properties-panel.update-businessobject-list',
+      cmd: 'element.updateModdleProperties',
       context: {
-        element: element,
-        currentObject: properties,
-        propertyName: 'values',
-        objectsToRemove: [ property ]
+        element,
+        moddleElement: properties,
+        properties: {
+          values
+        }
       }
     });
 
     // remove camunda:Properties if there are no properties anymore
-    if (properties.get('values').length === 1) {
+    if (!values.length) {
+      const businessObject = getBusinessObject(element),
+            extensionElements = businessObject.get('extensionElements');
+
+      values = without(extensionElements.get('values'), properties);
+
       commands.push({
-        cmd: 'properties-panel.update-businessobject-list',
+        cmd: 'element.updateModdleProperties',
         context: {
-          element: element,
-          currentObject: getBusinessObject(element).get('extensionElements'),
-          propertyName: 'values',
-          objectsToRemove: [ properties ]
+          element,
+          moddleElement: extensionElements,
+          properties: {
+            values
+          }
         }
       });
+
+      if (!values.length) {
+        commands.push({
+          cmd: 'element.updateModdleProperties',
+          context: {
+            element,
+            moddleElement: businessObject,
+            properties: {
+              extensionElements: undefined
+            }
+          }
+        });
+      }
     }
 
     commandStack.execute('properties-panel.multi-command-executor', commands);
```

Comment:
Hmm if we really migrate to only use core modeling API, we should also get rid of the multi command thing.

Reply:
Same here. I wanted to keep the changes simple. 🤡

Reply:
Which is fine.